app/lib/actions/katello/content_view/promote.rb

```diff
@@ -10,6 +10,7 @@ def plan(version, environments, is_force = false, description = nil, options = {
 
           fail ::Katello::HttpErrors::BadRequest, _("Cannot promote environment out of sequence. Use force to bypass restriction.") if !is_force && !version.promotable?(environments)
 
+          plan_self(environments: environments.map(&:name))
```

Comment:
I just realized that someone looking at this may have no idea why this is here, think it does nothing and delete it.  Mind adding a comment here to explain?